### PR TITLE
Master+mt

### PR DESCRIPTION
**Pull request by vipear**
Cleaned up unused test code.
Adjusted forced release time constant and bias to make time ~5ms from start of forced release to attack. This seems to improve fast sequencing.
Changed noteOn and NoteOff to use switch-case statements.

### CODE REVIEW

**Comment by ElectroTechnique** [comment]
Works fine for me.

File: TSynth/VoiceGroup.h [hunk 1]
```diff
@@ -1,1018 +1,1018 @@
...
+    void noteOff(uint8_t note) {
+        if (this->monophonic) {
+            handleMonophonicNoteOff(note);
+            return;
+        }
+
+        uint8_t num = 0;
+        switch (this->_params.unisonMode) {
+            case 0:
+                num = noteOff(note, false); // changed for experiment
```
Comment (winder) [inline TSynth/VoiceGroup.h:776]: Comment no longer applicable.

**Comment by ElectroTechnique** [comment]
Can you test the envelope with _Linear_ again, please? There's noise, which I haven't noticed before. I think it's from the last code update, as the current V2.21 release is fine. Thanks.

**Comment by vipear** [comment]
Rats. I should have checked the linear mode out. There was missing code in noteOn() line 64. It seems OK now.

    case STATE_IDLE_NEXT:
      count=delay_count;
      if(count>0)
      {
        state=STATE_DELAY;
        inc_hires=0;
      }
      else
      {
        state=STATE_ATTACK;
        count=attack_count;
        inc_hires=0x40000000/(int32_t)count;  //***************************** was missing the divisor (int32_t)count;
      }

**Comment by winder** [comment]
In the else count is <= 0, wouldn't this cause a div by 0 error?

**Comment by vipear** [comment]
Count is reloaded with the attack count in the else statement. This would cause a division by zero problem if an attack time of less than 200.3 us was previously set. A divide by zero returns 0 on the Teensy 4.0 from my testing and assuming a divide by 0 exception is not set to halt normal program flow the state machine would just move to the hold state and continue.
However, be more robust this should be caught in the attack time set function

  FLASHMEM void attack(float milliseconds) {
    attack_count = milliseconds2count(milliseconds);
    if(attack_count==0) attack_count=1; //************* added check for div/0
    updateExpAttack();
  }

**Pull request merged**